### PR TITLE
Phase2-294B Extend functionality of RecHitTools  of HGCal and add a testing code

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Extend functionality of RecHitTools of HGCal  and add a testing code

#### PR validation:

Use a cfg file in RecoLocalCalo/HGCalRecAlgos/test

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: RecoLocalCalo/HGCalRecAlgos/test/BuildFile.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<use name="FWCore/MessageLogger"/>
+<use name="FWCore/ParameterSet"/>
+<use name="FWCore/Framework"/>
+<use name="Geometry/HGCalGeometry"/>
+<use name="Geometry/Records"/>
+<use name="RecoLocalCalo/HGCalRecAlgos"/>
+<flags EDM_PLUGIN="1"/>
+<library name="RecoLocalCaloHGCalRecAlgosPlugin" file="*.cc">
```
Comment (slava77) [inline RecoLocalCalo/HGCalRecAlgos/test/BuildFile.xml:8]:
```suggestion
<library name="RecoLocalCaloHGCalRecAlgosTestPlugin" file="*.cc">
```
?
to be more clear what this is

**Comment by bsunanda** [comment]
@cmsbuild please test with #36021

File: RecoLocalCalo/HGCalRecAlgos/test/BuildFile.xml (hunk 1 above)
Comment (bsunanda) [reply RecoLocalCalo/HGCalRecAlgos/test/BuildFile.xml]: Done that

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/36041	baa0447 
- code changes are essentially technical, in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

**Comment by srimanob** [comment]
+Upgrade

**Comment by perrotta** [comment]
+1

**Pull request merged**